datafusion/physical-plan/src/sorts/sort.rs

```diff
@@ -2402,4 +2454,757 @@ mod tests {
+            let a_curr = a_array.value(i);
+            let a_next = a_array.value(i + 1);
+            let b_curr = b_array.value(i);
+            let b_next = b_array.value(i + 1);
+
+            assert!(
+                a_curr < a_next || (a_curr == a_next && b_curr <= b_next),
+                "Not properly sorted at position {i}: ({a_curr}, {b_curr}) -> ({a_next}, {b_next})",
+            );
+        }
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn test_sort_batch_chunked_empty_batch() -> Result<()> {
+        let schema = Arc::new(Schema::new(vec![Field::new("a", DataType::Int32, false)]));
+
+        let batch = RecordBatch::new_empty(Arc::clone(&schema));
+
+        let expressions: LexOrdering =
+            [PhysicalSortExpr::new_default(Arc::new(Column::new("a", 0)))].into();
+
+        let result_batches = sort_batch_chunked(&batch, &expressions, 100)?;
+
+        // Empty input produces no output batches (0 chunks)
+        assert_eq!(result_batches.len(), 0);
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn test_sort_batch_chunked_single_row() -> Result<()> {
+        let schema = Arc::new(Schema::new(vec![Field::new("a", DataType::Int32, false)]));
+
+        let batch = RecordBatch::try_new(
+            Arc::clone(&schema),
+            vec![Arc::new(Int32Array::from(vec![42]))],
+        )?;
+
+        let expressions: LexOrdering =
+            [PhysicalSortExpr::new_default(Arc::new(Column::new("a", 0)))].into();
+
+        let result_batches = sort_batch_chunked(&batch, &expressions, 100)?;
+
+        assert_eq!(result_batches.len(), 1);
+        assert_eq!(result_batches[0].num_rows(), 1);
+        let array = as_primitive_array::<Int32Type>(result_batches[0].column(0))?;
+        assert_eq!(array.value(0), 42);
+
+        Ok(())
+    }
+
+    // ========================================================================
+    // Tests for get_reserved_byte_for_record_batch()
+    // ========================================================================
+
+    #[tokio::test]
+    async fn test_get_reserved_byte_for_record_batch_normal_batch() -> Result<()> {
+        let schema = Arc::new(Schema::new(vec![Field::new("a", DataType::Int32, false)]));
+        let batch = RecordBatch::try_new(
+            schema,
+            vec![Arc::new(Int32Array::from(vec![1, 2, 3, 4, 5]))],
+        )?;
+
+        let reserved = get_reserved_byte_for_record_batch(&batch)?;
+
+        // Should be greater than 0
+        assert!(reserved > 0);
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn test_get_reserved_byte_for_record_batch_with_sliced_batches() -> Result<()> {
+        let schema = Arc::new(Schema::new(vec![Field::new("a", DataType::Int32, false)]));
+
+        // Create a larger batch then slice it
+        let large_array = Int32Array::from((0..1000).collect::<Vec<i32>>());
+        let sliced_array = large_array.slice(100, 50); // Take 50 elements starting at 100
```

Comment:
this test doesn't really test that sliced array are counted

instead only take 1 item and check the reserved byte of the sliced and the reserved byte of the non sliced and assert that the non sliced is larger